### PR TITLE
Remove ineffective check from FluxMonitor/geth integration test

### PR DESCRIPTION
**Pull request by coventry**
This was caught by @vlyl's `staticcheck` work (Thanks, @vlyl.)  `reportPrice = answer + 1` was intended to have a side-effect, attempting to trigger a premature report from the fluxmonitor. For that to work, the `reportPrice` argument to `checkNewRoundBlocked` should have been a pointer. However, fluxmonitor is too well-behaved to fall for that, and the message this test was checking for, `"skipping poll: not eligible to submit"` occurs anyway, so this test was passing without checking anything useful.

### CODE REVIEW

File: core/adapters/http.go [hunk 1]
```diff
@@ -331,8 +331,7 @@ type QueryParameters url.Values
 
 // UnmarshalJSON implements the Unmarshaler interface
 func (qp *QueryParameters) UnmarshalJSON(input []byte) error {
-	values := url.Values{}
-	strs := []string{}
+	var strs = []string{}
```
Comment (coventry) [inline core/adapters/http.go:334]: You can just use `var strs []string`, here.

File: core/internal/gethwrappers/go_generate_test.go [hunk 2]
```diff
@@ -93,6 +93,7 @@ func compareCurrentCompilerAritfactAgainstRecordsAndSoliditySources(
 	// Normalize the whitespace in the ABI JSON
 	abiBytes, err := utils.NormalizedJSON(
 		[]byte(gjson.GetBytes(compilerJSON, abiPath).String()))
+	assert.NoError(t, err, "failed to normalized JSON %s", compilerJSON)
```
Comment (coventry) [inline core/internal/gethwrappers/go_generate_test.go:96]: `normalize` instead of  `normalized`.

File: core/store/models/user_test.go [hunk 3]
```diff
@@ -55,11 +55,13 @@ func TestAuthenticateUserByToken(t *testing.T) {
 	var user models.User
 
 	token, err := user.GenerateAuthToken()
+	assert.NoError(t, err, "failed when generate auth token")
```
Comment (coventry) [inline core/store/models/user_test.go:58]: `failed to generate auth token`

File: core/store/orm/orm_test.go [hunk 4]
```diff
@@ -1154,6 +1154,7 @@ func TestORM_FindTxByAttempt_CurrentAttempt(t *testing.T) {
 
 	createdTx := cltest.CreateTx(t, store, from, 1)
 	fetchedTx, fetchedTxAttempt, err := store.FindTxByAttempt(createdTx.Hash)
+	assert.NoError(t, err, "failed to find tx:%s from store", createdTx.Hash)
```
Comment (coventry) [inline core/store/orm/orm_test.go:1157]: `in store`

File: core/store/store.go [hunk 5]
```diff
@@ -107,9 +107,9 @@ func (wrapper *lazyRPCWrapper) Subscribe(ctx context.Context, channel interface{
 		return nil, err
 	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	newCtx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
```
Comment (coventry) [inline core/store/store.go:110]: I would instead change the parameter name `ctx` in the function signature to `_`, to indidcate that it's unused.

File: core/services/fluxmonitor/flux_monitor_simulated_blockchain_test.go [hunk 6]
```diff
@@ -228,7 +226,7 @@ func submitAnswer(t *testing.T, p answerParams) {
 }
 
 type maliciousFluxMonitor interface {
-	XXXTestingOnlyCreateJob(t *testing.T, jobSpecId *models.ID,
+	CreateJob(t *testing.T, jobSpecId *models.ID,
```
Comment (samsondav) [inline core/services/fluxmonitor/flux_monitor_simulated_blockchain_test.go:229]: This seems like a functional change...

File: core/services/fluxmonitor/helpers_test.go [hunk 7]
```diff
@@ -94,10 +94,10 @@ func dataWithResult(t *testing.T, result decimal.Decimal) adapterResponseData {
 	return data
 }
 
-// XXXTestingOnlyCreateJob is used in TestFluxMonitorAntiSpamLogic to create a
+// CreateJob is used in TestFluxMonitorAntiSpamLogic to create a
 // job with a specific answer and round, for testing nodes with malicious
 // behavior
-func (fm *concreteFluxMonitor) XXXTestingOnlyCreateJob(t *testing.T,
+func (fm *concreteFluxMonitor) CreateJob(t *testing.T,
```
Comment (samsondav) [inline core/services/fluxmonitor/helpers_test.go:100]: @coventry can you shed any light on this rename?

File: core/store/store.go [hunk 8]
```diff
@@ -101,7 +101,7 @@ func (wrapper *lazyRPCWrapper) Call(result interface{}, method string, args ...i
 	return wrapper.client.Call(result, method, args...)
 }
 
-func (wrapper *lazyRPCWrapper) Subscribe(ctx context.Context, channel interface{}, args ...interface{}) (eth.Subscription, error) {
+func (wrapper *lazyRPCWrapper) Subscribe(_ context.Context, channel interface{}, args ...interface{}) (eth.Subscription, error) {
```
Comment (samsondav) [inline core/store/store.go:104]: We probably shouldn't actually be throwing away the context... Can we use it instead?

Comment (vlyl) [reply core/store/store.go:104]: the ctx parameter was overwritten by a new context in function before it first be used, so the staticcheck warning this. For avoiding this warning, is it a better idea that leave the ctx parameter and use a newCtx instead?

File: core/services/fluxmonitor/helpers_test.go (hunk 7 above)
Comment (coventry) [reply core/services/fluxmonitor/helpers_test.go:100]: The `XXXTestingOnly` is a hold-over from when the simulated-blockchain test lived in `internal`. Now that it lives in `fluxmonitor`, the fact that it's in a `*_test.go` file forces that the function can only be used in tests.

File: core/store/store.go (hunk 8 above)
Comment (vlyl) [reply core/store/store.go]: use ctx in parameters instead of create new one in function.

**Pull request merged**